### PR TITLE
workaround: Optional email when login via github

### PR DESCRIPTION
**Pull request by jupe**
Allow to login with GitHub private email.

Solution is not robust, and probably needs better approach but covers basic scenarios.

Created top of #33

### CODE REVIEW

File: app/controllers/authentication.js [hunk 1]
```diff
@@ -20,9 +20,22 @@ class AuthenticationController {
   constructor() {
     this.config = {
       GOOGLE_SECRET: nconf.get('google_secret'),
-      TOKEN_SECRET: nconf.get('webtoken'),
+      TOKEN_SECRET: nconf.get('webtoken')
```
Comment (olkorhon) [inline app/controllers/authentication.js]: Trailing comma is an eslint preference, personally I do not like it but that is why it exists.

File: app/controllers/authentication.js [hunk 2]
```diff
@@ -20,9 +20,22 @@ class AuthenticationController {
...
+  
+  static generateEmail(name) {
+    // @todo more robust mechanism needed
+    // This assumes that name is "<first-name> (<second-name> )<last-name>"
+    // and EMAIL_DOMAIN is specified
+    let parts = _.map(name.split(" "), _.toLower);
+    if(parts.length<2) {
+      return '';
+    }
+    let email = `${parts[0]}.${parts[parts.length-1]}@${emailDomain}`;
```
Comment (olkorhon) [inline app/controllers/authentication.js]: Spaced infix operators, e.g. "parts.length - 1", we really should start enforcing a specified set of eslint rules. Personally I like the spaced infix operators as they make it easier to spot operators and subsequently split the line into logical blocks.

File: app/controllers/authentication.js [hunk 3]
```diff
@@ -20,9 +20,22 @@ class AuthenticationController {
...
+      TOKEN_SECRET: nconf.get('webtoken')
     };
   }
+  
+  static generateEmail(name) {
+    // @todo more robust mechanism needed
+    // This assumes that name is "<first-name> (<second-name> )<last-name>"
+    // and EMAIL_DOMAIN is specified
+    let parts = _.map(name.split(" "), _.toLower);
+    if(parts.length<2) {
```
Comment (olkorhon) [inline app/controllers/authentication.js]: Eslint by default prefers spaced infix operators, e.g. "parts.length < 2".

File: app/controllers/authentication.js (hunk 1 above)
Comment (jupe) [inline app/controllers/authentication.js]: we can create own rules which allows this. I don't like that either :)

File: app/controllers/authentication.js (hunk 2 above)
Comment (jupe) [inline app/controllers/authentication.js]: yep, we have to create eslint configuration to root folder and run eslint in PR check

**Comment by jupe** [comment]
> Removing mandatory email and searching users via name and/or github username could be a better option.

would be much better approach. Let's keep this open until we have such feature implemented and close this PR then without merging :)

**Comment by jupe** [comment]
Actually reason is this: https://stackoverflow.com/questions/35373995/github-user-email-is-null-despite-useremail-scope

url should be `/user/emails` which give we also private marked emails.

**Pull request closed without merging**